lms/djangoapps/instructor/tests/test_proctoring.py

```diff
@@ -36,24 +34,74 @@ def setUp(self):
+    def setUpCourseUrl(self, course):
+        """
+        Create URL for instructor dashboard
+        """
+        self.url = reverse('instructor_dashboard', kwargs={'course_id': course.id.to_deprecated_string()})
 
-    def test_pass_proctoring_tab_in_instructor_dashboard(self):
+    def setUpCourse(self, enable_proctored_exams, enable_timed_exams):
         """
-        Test Pass Proctoring Tab is in the Instructor Dashboard
+        Create course based on proctored exams and timed exams values
         """
+        self.course = CourseFactory.create(enable_proctored_exams=enable_proctored_exams,
+                                           enable_timed_exams=enable_timed_exams)
+        self.setUpCourseUrl(self.course)
+
+    @ddt.data(
+        (True, False),
+        (False, True)
+    )
+    @ddt.unpack
+    def test_proctoring_tab_visible_for_global_staff(self, enable_proctored_exams, enable_timed_exams):
+        """
+        Test Proctoring Tab is visible in the Instructor Dashboard
+        for global staff
+        """
+        self.setUpCourse(enable_proctored_exams, enable_timed_exams)
+
         self.instructor.is_staff = True
         self.instructor.save()
 
-        response = self.client.get(self.url)
-        self.assertIn(self.proctoring_link, response.content)
-        self.assertIn('Allowance Section', response.content)
+        # expects that proctoring tab is visible for global staff
+        self._assert_proctoring_tab_is_available()
+
+    @ddt.data(
+        (True, False),
+        (False, True)
+    )
+    @ddt.unpack
+    def test_proctoring_tab_visible_for_course_staff_and_admin(self, enable_proctored_exams, enable_timed_exams):
+        """
+        Test Proctoring Tab is visible in the Instructor Dashboard
+        for course staff(role of STAFF or ADMIN)
+        """
+        self.setUpCourse(enable_proctored_exams, enable_timed_exams)
 
-    def test_no_tab_non_global_staff(self):
+        self.instructor.is_staff = False
+        self.instructor.save()
+
+        # expects that proctoring tab is visible for course staff
+        CourseStaffRole(self.course.id).add_users(self.instructor)
+        self._assert_proctoring_tab_is_available()
+
+        # expects that proctoring tab is visible for course instructor
+        CourseStaffRole(self.course.id).remove_users(self.instructor)
+        CourseInstructorRole(self.course.id).add_users(self.instructor)
+        self._assert_proctoring_tab_is_available()
+
+    @ddt.data(
+        (True, False),
+        (False, True)
+    )
+    @ddt.unpack
+    def test_no_proctoring_tab_non_global_staff(self, enable_proctored_exams, enable_timed_exams):
         """
-        Test Pass Proctoring Tab is not in the Instructor Dashboard
-        for non global staff users
+        Test Proctoring Tab is not visible in the Instructor Dashboard
+        for course team other than course staff and course admin
```

Comment:
I didn't get what message we are intended to deliver here? course team vs course staff? what the difference?

Reply:
Course team has different roles. Staff and Admin are two roles of course team. I'll update the docstring.